bridge/github/export_test.go

```diff
@@ -0,0 +1,362 @@
+		},
+		&testCase{
+			name:    "bug with comments",
+			bug:     bugWithComments,
+			numOrOp: 2,
+		},
+		&testCase{
+			name:    "bug label change",
+			bug:     bugLabelChange,
+			numOrOp: 4,
+		},
+		&testCase{
+			name:    "bug with comment editions",
+			bug:     bugWithCommentEditions,
+			numOrOp: 4,
+		},
+		&testCase{
+			name:    "bug changed status",
+			bug:     bugStatusChanged,
+			numOrOp: 3,
+		},
+		&testCase{
+			name:    "bug title edited",
+			bug:     bugTitleEdited,
+			numOrOp: 2,
+		},
+	}, nil
+
+}
+
+func TestExporter(t *testing.T) {
+	user := os.Getenv("TEST_USER")
+	token := os.Getenv("GITHUB_TOKEN_ADMIN")
+	if token == "" {
+		t.Skip("Env var GITHUB_TOKEN_ADMIN missing")
+	}
+
+	repo := repository.CreateTestRepo(false)
+	defer repository.CleanupTestRepos(t, repo)
+
+	backend, err := cache.NewRepoCache(repo)
+	require.NoError(t, err)
+
+	author, err := backend.NewIdentity("test identity", "test@test.org")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = backend.SetUserIdentity(author)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer backend.Close()
+	interrupt.RegisterCleaner(backend.Close)
+
+	tests, err := testCases(backend, author)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// generate project name
+	projectName := generateRepoName()
+
+	// create repository
+	if err := createRepository(projectName, token); err != nil {
+		t.Fatal(err)
+	}
+	fmt.Println("created repository", projectName)
+
+	// delete repository before ending tests
+	defer func(t *testing.T) {
+		if err := deleteRepository(projectName, user, token); err != nil {
+			t.Fatal(err)
+		}
+		fmt.Println("deleted repository:", projectName)
+	}(t)
+
+	// initialize exporter
+	exporter := &githubExporter{}
+	err = exporter.Init(core.Configuration{
+		keyOwner:   user,
+		keyProject: projectName,
+		keyToken:   token,
+	})
+	require.NoError(t, err)
+
+	start := time.Now()
+
+	// export all bugs
+	err = exporter.ExportAll(backend, time.Time{})
+	require.NoError(t, err)
+
+	fmt.Printf("test repository exported in %f seconds\n", time.Since(start).Seconds())
+
+	repoTwo := repository.CreateTestRepo(false)
+	defer repository.CleanupTestRepos(t, repoTwo)
+
+	// create a second backend
+	backendTwo, err := cache.NewRepoCache(repoTwo)
+	require.NoError(t, err)
+
+	importer := &githubImporter{}
+	err = importer.Init(core.Configuration{
+		keyOwner:   user,
+		keyProject: projectName,
+		keyToken:   token,
+	})
+	require.NoError(t, err)
+
+	// import all exported bugs to the second backend
+	err = importer.ImportAll(backendTwo, time.Time{})
+	require.NoError(t, err)
+
+	require.Len(t, backendTwo.AllBugsIds(), len(tests))
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// for each operation a SetMetadataOperation will be added
+			// so number of operations should double
+			require.Len(t, tt.bug.Snapshot().Operations, tt.numOrOp*2)
+
+			for _, op := range tt.bug.Snapshot().Operations {
+				if _, ok := op.(*bug.SetMetadataOperation); !ok {
```

Comment:
```suggestion
				// Check if the originals operations (*not* SetMetadata) are tagged properly
				if _, ok := op.(*bug.SetMetadataOperation); !ok {
```

It's very subtle that you are actually *not* checking the SetMetadataOperation ops